tcms/static/js/utils.js

```diff
@@ -1,17 +1,32 @@
 /*
     Used to update a select when something else changes.
 */
-function updateSelect(data, selector, id_attr, value_attr) {
+function updateSelect(data, selector, id_attr, value_attr, group_attr) {
     var _select_tag = $(selector)[0];
     var new_options = '';
 
-    // in some cases, e.g. TestRun search, the 1st <option> element is ---
+	currentGroup = '';
+	// check if using multiple select
+	isMultiple = _select_tag.attributes.getNamedItem('multiple') !== null;
+    
+	// in some cases using single select, the 1st <option> element is ---
     // which must always be there to indicate nothing selected
-    if (_select_tag.options.length) {
+    if (!isMultiple && _select_tag.options.length) {
         new_options = _select_tag.options[0].outerHTML;
     }
+	
+	data.forEach(function(element) {
+		if(isMultiple) {
+			if (group_attr != null && currentGroup !== element[group_attr]) {
```

Comment:
Given your explanation in https://github.com/kiwitcms/Kiwi/pull/2335#discussion_r628761755 I think these 2 if statements can be merged into 1.

Reply:
I thought it would be easier for future maintainability if we have an outer isMultiple block where we can put all multiple select logic. Then the optgroup block would be a subsection of that block. Should I still make this change?

Reply:
I'm not concerned about future maintainability ATM but rather simplicity and the code being easy to read. That said I have to see how things look like before making up my mind.  IMO go for a single if condition and if necessary in the future that can be changed very easily.